ghcide/src/Development/IDE/Main.hs

```diff
@@ -187,18 +235,16 @@ data Arguments = Arguments
     , argsThreads               :: Maybe Natural
     }
 
-instance Default Arguments where
-    def = defaultArguments Info
 
-defaultArguments :: Priority -> Arguments
-defaultArguments priority = Arguments
+defaultArguments :: Recorder (WithPriority Log) -> Logger -> Arguments
+defaultArguments recorder logger = Arguments
         { argsProjectRoot = Nothing
         , argsOTMemoryProfiling = False
         , argCommand = LSP
-        , argsLogger = stderrLogger priority
-        , argsRules = mainRule def >> action kick
+        , argsLogger = pure logger
```

Comment:
What is the old `Logger` type used for now? If it is still being used, can it be subsumed with a `LogLegacy Priority` recorder?

Reply:
The old logger type is still being used, it's just being initialized at various entry points and passed in.

Reply:
Not exactly sure what the `LogLegacy Priority` means, but the original plan was to remove `argsLogger` completely eventually.